File: java/arcade/AbsoluteValuesSumMinimization.java
public final class AbsoluteValuesSumMinimization {

	int absoluteValuesSumMinimization(int[] a) {
	    int[] n = new int[a.length];
	    for(int i = 0; i < a.length; i++) {
	        for(int j : a) {
	            n[i] += Math.abs(j - a[i]);
	        }
	    }
	    int lower = n[0];
	    for(int i : n) { lower = (i < lower) ? i : lower; }
	    return a[Arrays.binarySearch(n, lower)];
	}
}

File: ruby/arcade/absoluteValuesSumMinimization.rb
def absoluteValuesSumMinimization(a)
	n = [], lower = 0
	a.size.times do |i|
		n[i] = 0
		a.each do |j|
			n[i] += (j - a[i]).abs
		end
	end
	lower = n.min
	return a[n.index(lower)]
end